Added 2 new effects, added asound with mic source, improvements in ex…

alexanderFell opened the pull request:
Added 2 new effects, added asound with mic source, improvements in existing effects

Added new effects:
1) strobe which makes all LEDs blink, once a threshold has been crossed. Since I currently use a
microphone attached to an external USB soundcard, I can control the sensitivity using alsamixer (Capture)
2) The effect "allEffects" checks the config file for existing effects and can switch among them randomly.

Added asound.conf with mic source
I added another asound.conf file, which uses the microphone attached to an external USB sound card (hw:1,0)
The user can currently choose among them, depending on the needs.

I also removed .asound.conf, since the normal user does not see it usually.

Improvements in existing effects:
VUMeter:
- Added a check, if the numbers of LEDs on the left and right side differ.
- Ensure that the effect does not crash on exit

all existing effects:
Added a check, if the module is invoked by another module (e.g. allEffects) or the main

main.py:
Now the capitalization of the exit command does not matter. I tend to write capital X, do not
know why.

Alex

alexanderFell committed:
Added 2 new effects, added asound with mic source, improvements in existing effects

Added new effects:
1) strobe which makes all LEDs blink, once a threshold has been crossed. Since I currently use a
microphone attached to an external USB soundcard, I can control the sensitivity using alsamixer (Capture)
2) The effect "allEffects" checks the config file for existing effects and can switch among them randomly.

Added asound.conf with mic source
I added another asound.conf file, which uses the microphone attached to an external USB sound card (hw:1,0)
The user can currently choose among them, depending on the needs.

I also removed .asound.conf, since the normal user does not see it usually.

Improvements in existing effects:
VUMeter:
+ Added a check, if the numbers of LEDs on the left and right side differ.
+ Ensure that the effect does not crash on exit

all existing effects:
Added a check, if the module is invoked by another module (e.g. allEffects) or the main

main.py:
Now the capitalization of the exit command does not matter. I tend to write capital X, do not
know why.

Alex

RanzQ commented:
@alexanderFell thanks for the PR. Could you update the README a bit though? Cause now it conflicts a bit with the changes you made (no more .asoundrc)

alexanderFell committed:
Absolute paths for effects and configuration file reading

The main.py can be called from other directories as absolute paths independent of the location of the Pyhton script, are used now.

alexanderFell committed:
Adopted README.md to include instructions for the asound.conf

alexanderFell committed:
Fexed misspelling about asoundrc in the $HOME directory.

alexanderFell commented:
@RanzQ Readme.md adapted. Please have a look over it and let me know, if it fine. In addition, I added absolute paths to the main.py, so that I can call it from anywhere. In fact I wrote a little HTTP server for that (and more). (Enhancement #13)

Btw, the PR will remove bug #11 as well.

RanzQ commented:
It looks good, I'll test the new effects and merge. I've got something broken on my dev machine atm so wasn't able to launch this (even my own master).

RanzQ commented:
I tried the "strobe" effect with two of my configs (only using `--gui`, got no leds attached to this machine), first one blinked only one led like this:

![image](https://cloud.githubusercontent.com/assets/1150613/18408879/c44f7848-7744-11e6-8d7f-7986675f8848.png)

Other config has less leds but it throws an error:

```Exception in Tkinter callback
Traceback (most recent call last):
  File "/usr/lib/python2.7/lib-tk/Tkinter.py", line 1535, in __call__
    return self.func(*args)
  File "/usr/lib/python2.7/lib-tk/Tkinter.py", line 586, in callit
    func(*args)
  File "/home/ranzq/workspace/hyperion-audio-effects/app/gui.py", line 60, in update_leds
    change_color(canvas, led, i)
  File "/home/ranzq/workspace/hyperion-audio-effects/app/gui.py", line 70, in change_color
    b = hyperion.get_led_data()[3 * led_number + 2]
IndexError: bytearray index out of range
```

Both of my effects work but this might be a bug in my code too. I can upload the configs if you need for testing.

RanzQ commented on effects/strobe.py:
```diff
@@ -0,0 +1,122 @@
+
+        self._magnitudes = None
+
+
```
You've got a lot of mixed tabs and spaces:

![image](https://cloud.githubusercontent.com/assets/1150613/18408916/a99c7496-7745-11e6-9c11-1eff86e1de23.png)

I haven't used `pylint` on this project so there are a lot of errors if you would enable it, but mixed tabs and spaces are ugly on any code.

RanzQ commented:
I added a `.pylintrc` now. Please rebase your fork and check all your python files with [pylint](https://docs.pylint.org/en/latest/tutorial.html). For example `pylint effects/strobe.py -rn`. Or use a SublimeLinter plugin like me.

Let me know if there are some rules that are too annoying, disabled ones are in `.pylintrc` and you can check the full list using `pylint --list-msgs`.

RanzQ commented:
Closing this for due to inactivity.
